pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -61,8 +61,11 @@ def on_trainer_init(
         self.trainer._is_data_prepared = False
 
     def get_profiled_train_dataloader(self, train_dataloader) -> Iterable:
-        self.data_fetcher = DataFetcher()
-        self.data_fetcher.setup(train_dataloader)
+        # FIXME: Temporary hack
```

Comment:
why is this temporary? Do you intend to fix it here?

Reply:
Because there is no way to select `InterBatchParallelDataFetcher`.

This is set directly within the test as 

The next PR will clean this.

```py
trainer.data_connector.data_fetcher = InterBatchParallelDataFetcher()
```